positron/modules/process.js

```diff
@@ -16,8 +16,15 @@ Cu.import('resource://gre/modules/Services.jsm');
 
 const exeFile = Services.dirsvc.get("XREExeF", Ci.nsIFile);
 
-// Re-export `process` as a native module.  I dunno why this matters, since all
-// modules already have access to the global.  But Node does it, so we do too.
+// This file is a special kind of module, as the module loader requires it
+// during loader construction and uses it to define the `process` global.
+// Thus that global is not available at evaluation time, neither for this module
+// nor for any other modules it requires upon evaluation.
+
+const EventEmitter = require('events').EventEmitter;
+const process = new EventEmitter();
```

Comment:
> If the above is correct, we should really refactor the process object to define its properties in adherence to Object.create().

I'm unsure what this means, but the intention here is to create a *process* singleton that extends the *EventEmitter* base functionality with the *process* object's additional properties and methods.

I could create a *Process* class or prototype and use classical or prototypal inheritance to establish the relationship between *EventEmitter* and that class/prototype, and that would make sense if I was constructing multiple instances of the *process* object, but it seems like overkill when *process* is a singleton.